boringtun/src/device/integration_tests/mod.rs

```diff
@@ -633,6 +634,48 @@ mod tests {
         assert_eq!(response, encode(PublicKey::from(&peer.key).as_bytes()));
     }
 
+    /// Test if wireguard can handle simple ipv4 connections with no traffic for over REKEY_TIMEOUT
+    #[test]
+    #[ignore]
+    fn test_wg_start_ipv4_traffic_flows_after_rekey_timeout() {
+        let port = next_port();
+        let private_key = StaticSecret::new(OsRng);
+        let public_key = PublicKey::from(&private_key);
+        let addr_v4 = next_ip();
+        let addr_v6 = next_ip_v6();
+
+        let mut wg = WGHandle::init(addr_v4, addr_v6);
+
+        assert_eq!(wg.wg_set_port(port), "errno=0\n\n");
+        assert_eq!(wg.wg_set_key(private_key), "errno=0\n\n");
+
+        // Create a new peer whose endpoint is on this machine
+        let mut peer = Peer::new(
+            SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), next_port()),
+            vec![AllowedIp {
+                ip: next_ip(),
+                cidr: 32,
+            }],
+        );
+
+        peer.start_in_container(&public_key, &addr_v4, port);
+
+        let peer = Arc::new(peer);
+
+        wg.add_peer(Arc::clone(&peer));
+        wg.start();
+
+        let response = peer.get_request();
+
+        assert_eq!(response, encode(PublicKey::from(&peer.key).as_bytes()));
+
+        sleep(Duration::from_secs(120 + 5));
```

Comment:
Are you sure you meant `REKEY-AFTER-TIME` instead of `REJECT-AFTER-TIME`? https://www.wireguard.com/papers/wireguard.pdf#subsection.6.1

Reply:
I actually had `Reject-After-Time` there but decided that maybe this in an overkill... since you also noticed this, I will just set it back to 180+5